l10n_ch_base_bank/models/bank.py

```diff
@@ -1,86 +1,62 @@
-# Copyright 2012-2017 Camptocamp
+# Copyright 2012-2019 Camptocamp
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 import re
 from odoo import models, fields, api, _
 from odoo.tools import mod10r
-from odoo import exceptions
+from odoo.exceptions import ValidationError
 
-from odoo.addons.base_iban.models.res_partner_bank import normalize_iban
-
-# bic for all switzerland's post offices is defined
+CH_POSTFINANCE_CLEARING = "09000"
 CH_POST_BIC = 'POFICHBEXXX'
 
 
-class BankCommon(object):
+def validate_l10n_ch_postal(postal_acc_number):
+    """Check if the string postal_acc_number is a valid postal account number,
+    i.e. it only contains ciphers and is last cipher is the result of a
+    recursive modulo 10 operation ran over the rest of it. Shorten form with -
+    is also accepted.
+    Raise a ValidationError if check fails
+    """
+    if not postal_acc_number:
+        raise ValidationError(_("There is no postal account number."))
+    if re.match('^[0-9]{2}-[0-9]{1,6}-[0-9]$', postal_acc_number or ''):
```

Comment:
do you need 'or' here?

Reply:
You are right, there is already a check on `not postal_acc_number` thus the or is useless.

Reply:
Fixed